src/components/button/button.d.ts

```diff
@@ -1,34 +1,54 @@
 import * as React from "react";
 import { SpacingProps } from "../../utils/helpers/options-helper";
-import { IconTypes, ButtonTypes } from "../../utils/helpers/options-helper/options-helper";
+import {
+  IconTypes,
+  ButtonTypes,
+} from "../../utils/helpers/options-helper/options-helper";
 
 export interface ButtonProps extends SpacingProps {
+  /** Prop to specify the aria-label text.
+   *  Only to be used in Button when only an icon is rendered.
+   * This is required to comply with WCAG 4.1.2 - Buttons must have discernible text
+   */
+  "aria-label"?: string;
+  /** [Legacy] Button types for legacy theme: "primary" | "secondary" */
   as?: ButtonTypes;
+  /** Color variants for new business themes: "primary" | "secondary" | "tertiary" | "darkBackground" */
   buttonType?: ButtonTypes;
-  "aria-label"?: string;
+  /** The text the button displays */
+  children?: React.ReactNode;
+  /** Apply disabled state to the button */
   disabled?: boolean;
+  /** Apply destructive style to the button */
   destructive?: boolean;
+  /** Ref to be forwarded */
+  forwardRef?: React.Ref<HTMLButtonElement>;
+  /** Apply fullWidth style to the button */
   fullWidth?: boolean;
-  /** Margin bottom, given number will be multiplied by base spacing unit (8) */
-  mb?: 0 | 1 | 2 | 3 | 4 | 5 | 7;
-  /** Margin left, any valid CSS value */
-  ml?: string;
-  size?: "small" | "medium" | "large";
-  iconPosition?: "before" | "after";
-  iconType?: IconTypes;
-  subtext?: string;
-  children?: React.ReactNode;
   /** Used to transform button into anchor */
   href?: string;
-  forwardRef?: () => void;
-  onClick?: (event: React.MouseEvent<HTMLButtonElement | HTMLLinkElement>) => void;
```

Comment:
There is an eslint error on the onClick prop  for the button component: 
` Property 'onClick' does not exist on type 'IntrinsicAttributes & ButtonProps'`